test/verify/check-packagekit

```diff
@@ -231,6 +240,179 @@ class TestUpdates(NoSubManCase):
         self.assertEqual(b.attr(self.update_icon, "class"), "fa fa-check-circle-o")
         b.wait_text(self.update_text, "System is up to date")
 
+    def testTracer(self):
+        b = self.browser
+        m = self.machine
+
+        class Tracer(object):
+            def __init__(self, testObj, rebootRequired=False, testStatusCard=False, packageName="vanilla", appName="foo", serviceName="slumber"):
+                self.testObj = testObj
+                self.rebootRequired = rebootRequired
+                self.testStatusCard = testStatusCard
+                self.packageName = packageName
+                self.appName = appName
+                self.serviceName = serviceName
+
+            def execute(self):
+                self.prepare()
+                self.test_frontend()
+                self.verify_backend()
+                self.cleanup()
+
+            def prepare(self):
+                m.execute("mkdir -p /etc/tracer")
+                self.testObj.write_file("/usr/local/bin/{0}".format(self.appName), "#!/bin/sh\nsleep infinity")
+                m.execute("chmod 744 /usr/local/bin/{0}".format(self.appName))
+                self.testObj.write_file("/etc/systemd/system/{0}.service".format(self.serviceName),
+                        """
+[Service]
+ExecStart=/usr/local/bin/{0}
+""".format(self.appName))
+                if self.rebootRequired:
+                    self.testObj.write_file("/etc/tracer/applications.xml",
+"""
+<applications>
+   <app name="{0}" type="static" />
+</applications>
```

Comment:
What does that do? Tell tracer to consider changes of these applications as "needs reboot"?

Reply:
Exactly, https://tracer.readthedocs.io/en/latest/user-guide/#helpers